src/App.tsx

```diff
@@ -1,14 +1,73 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+
+import { Todo } from './types/Todo';
+
+type Params = {
+  query: string;
+  filter: string;
+};
+
+function filterTodos(list: Todo[], params: Params) {
+  const { query, filter } = params;
+  let result: Todo[] = [];
+
+  if (filter) {
+    switch (filter) {
+      case 'all':
+        result = list;
+        break;
+
+      case 'active':
+        result = list.filter((todo: Todo) => !todo.completed);
+        break;
+
+      case 'completed':
+        result = list.filter((todo: Todo) => todo.completed);
+        break;
+
+      default:
+        result = list;
+        break;
+    }
+  }
+
+  if (query) {
+    result = result.filter((todo: Todo) =>
+      todo.title.toLowerCase().includes(query.toLowerCase()),
+    );
+  }
+
+  return result;
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+
+  const [select, setSelect] = useState<number | undefined>();
```

Comment:
The `select` state is initialized with `undefined`. Ensure that this state is used consistently and its role is well-defined across the application to avoid issues with displaying the correct todo in the modal.